src/main/scala/scala/tools/refactoring/sourcegen/TreeChangesDiscoverer.scala

```diff
@@ -156,9 +156,16 @@ trait TreeChangesDiscoverer {
           List((changed, orig.pos, Set(changed) ++ searchChildrenForChanges(changed)))
         }
 
+        def drillDownChildrenAndCollectChangesWithPosition(parent: Tree): List[(Tree, Position, Set[Tree])] =
```

Comment:
```scala
class A {
  def foo = {
    import acme.B
    import acme.S
    def bar = {
        import emca.D
        import emca.C
        ....
    }
  }
}
```
in that example changes were collected from `foo` only although the tree has been updated for `bar` as well. I'm not sure if this is the best place for fix but definitely it required any action.